src/main/java/org/truffleruby/core/encoding/EncodingNodes.java

```diff
@@ -860,9 +860,9 @@ private void raiseException(Object first, Object second) {
                 toRubyEncodingNode = insert(ToRubyEncodingNode.create());
             }
 
-            throw new RaiseException(getContext(), coreExceptions().encodingCompatibilityErrorIncompatible(
-                    toRubyEncodingNode.executeToEncoding(first),
+            throw new RaiseException(getContext(), coreExceptions().encodingCompatibilityErrorRegexpIncompatible(
                     toRubyEncodingNode.executeToEncoding(second),
+                    toRubyEncodingNode.executeToEncoding(first),
```

Comment:
It's not quite correct to use this message for all usages of CheckEncodingNode.
Probably `TruffleRegexpNodes.PrepareRegexpEncodingNode` should be used for places checking regexp encoding compatible with string encoding.

Reply:
I updated it here: https://github.com/oracle/truffleruby/pull/3107/files#diff-83dfc9322e25ef633ef3f69c48ba1dbc39b31fa42695ec8873cb98d68f28ebdcR837 . Any reason why this is might not be the right place?

Reply:
I think Regexp+String encoding negotiation is slightly different, and that shows via the message.
`if (first instanceof RubyRegexp || second instanceof RubyRegexp) {` might work, but it hides the problem we are using the wrong way to noegotiate Regexp+String encoding.
Also sometimes the Regexp is the 1st arg and sometimes 2nd, but the message should always have them in the right order.

Reply:
What would be great is to have specs for error messages for all usages of `CheckEncodingNode`, so we are sure to be correct.